### PR TITLE
Add Bison to Dockerimage files

### PR DESCRIPTION
**Pull request by k0ekk0ek**
Like with this [PR](https://github.com/ros-infrastructure/ros2-cookbooks/pull/29), the reason is described in more detail [here](https://github.com/ros2/ros2_documentation/pull/1191).

Please consider pulling the changes. Thanks!

### CODE REVIEW

File: linux_docker_resources/Dockerfile
Comment (nuclearsandwich) [inline linux_docker_resources/Dockerfile:111]:
```suggestion
RUN apt-get update && apt-get install --no-install-recommends -y bison libcunit1-dev maven openjdk-11-jdk
```

Let's preserve the lexical sorting.

File: linux_docker_resources/Dockerfile-RHEL
Comment (nuclearsandwich) [inline linux_docker_resources/Dockerfile-RHEL:46]: Can you please move the added package to preserve lexical sorting.

**Comment by nuclearsandwich** [comment]
@k0ekk0ek if you make the requested changes and pushes from maintainers are enabled I'll go ahead and update the ros2-cookbooks submodule in this PR so we can test CentOS, Ubuntu, and Windows with this one PR.

**Comment by k0ekk0ek** [comment]
@nuclearsandwich, thanks for reviewing. I've updated the PR to preserve lexical ordering.

**Comment by k0ekk0ek** [comment]
As per instructions on [PR 26 for ros2-cookbooks](https://github.com/ros-infrastructure/ros2-cookbooks/pull/29), I should update the submodule. However, the submodule has the branch `no-qt-maintenance-app` listed. Should that branch be updated to include Bison too, or should we just use the `latest` branch instead?

**Review by nuclearsandwich — APPROVED** [review]

**Comment by nuclearsandwich** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13971/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8802/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11670/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14054/)

**Comment by nuclearsandwich** [comment]
Okay it looks like there are merge conflicts between the default and `xtypes` branches of cyclonedds.

I've run a second set of builds that does not use the xtypes branch of cyclonedds to just validate that this environment change doesn't cause regressions in the cyclone mainline.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13972/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8803/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11671/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14055/)

**Comment by clalancette** [comment]
CI passed with this branch and the default (master) branch of CycloneDDS.  Thus I think this change is safe.  I'm going to go ahead and merge this, which should unblock @k0ekk0ek

**Pull request merged**
